File: Cpp/Number_of_Substrings_With_Only_1s.cpp
class Solution {
public:
    int numSub(string s) {
        int mod = 1e9 + 7;
        long cnt = 0;
        int curr = 0;
        for (char c : s) {
            if (c == '1') {
                curr += 1;
                cnt += curr;
            } else
                curr = 0;
        }
        return cnt % mod;
    }
};

File: Rust/Number_of_Substrings_With_Only_1s.rs
// time complexity: O(n)
// space complexity: O(1)

impl Solution {
    pub fn num_sub(s: String) -> i32 {
        const MOD: i64 = 1000000007;
        let mut cnt: i64 = 0;
        let mut curr = 0;
        for c in s.chars() {
            if c == '1' {
                curr += 1;
                cnt += curr;
            } else {
                curr = 0;
            }
        }
        (cnt % MOD).try_into().unwrap()
    }
}
